### PR TITLE
Add MPB recipe

### PR DESCRIPTION
**Pull request by ChristopherHogan**

### CODE REVIEW

**Comment by ChristopherHogan** [comment]
This is passing all tests and ready for review.
@conda-forge/core

**Comment by scopatz** [comment]
Thanks @ChristopherHogan!

File: recipes/mpb/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - openblas
+    - gcc
+    - fftw
+    - hdf5
```
Comment (jakirkham) [inline recipes/mpb/meta.yaml]: So `hdf5` should be pinned to `1.10.1`.

File: recipes/mpb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  url: https://github.com/stevengj/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - openblas
```
Comment (jakirkham) [inline recipes/mpb/meta.yaml]: `openblas` should be pinned to `0.2.20|0.2.20.*`.

File: recipes/mpb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - gcc
+    - fftw
+    - hdf5
+    - libctl >=4.0.0
+  run:
+    - openblas
+    - fftw
+    - hdf5
```
Comment (jakirkham) [inline recipes/mpb/meta.yaml]: Also `1.10.1` here as well.

File: recipes/mpb/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  build:
+    - openblas
+    - gcc
+    - fftw
+    - hdf5
+    - libctl >=4.0.0
+  run:
+    - openblas
```
Comment (jakirkham) [inline recipes/mpb/meta.yaml]: Also `0.2.20|0.2.20.*.` here as well.

Comment (scopatz) [reply recipes/mpb/meta.yaml]: @jakirkham - I thought this was something we got with conda-build 3 and the conda-forge-pinnings

**Comment by ChristopherHogan** [comment]
Pinning the hdf5 and openblas versions in conda-forge/mpb-feedstock#1 is causing the build to fail on `anaconda upload` (which it shouldn't be trying to do for a PR, right?)
```bash
Traceback (most recent call last):
  File "/opt/conda/bin/upload_or_check_non_existence", line 129, in <module>
    main()
  File "/opt/conda/bin/upload_or_check_non_existence", line 119, in main
    upload(cli, fname, owner, channel)
  File "/opt/conda/bin/upload_or_check_non_existence", line 66, in upload
    env=os.environ)
  File "/opt/conda/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['anaconda', '--quiet', '-t', '/tmp/tmp33jnbvck/binstar.token', 'upload', '/home/conda/feedstock_root/build_artifacts/linux-64/mpb-1.6.2-1.tar.bz2', '--user=conda-forge', '--channel=main']' returned non-zero exit status 1.
+ exit 1
Exited with code 1
```

**Comment by jakirkham** [comment]
Please see [my comment on that PR]( https://github.com/conda-forge/mpb-feedstock/pull/1#issuecomment-384475948 ).

File: recipes/mpb/meta.yaml (hunk 4 above)
Comment (msarahan) [reply recipes/mpb/meta.yaml]: I mostly agree here, but not quite.

I'd advise an alternate pattern. In the build deps, remove all pins. Conda-build 3 will do the right thing here.

In the run deps, you won't get automatic pinning until the upstreams are edited to include run_exports and rebuilt. So, for those, you should do:

- {{ pin_compatible('openblas', max_pin='x.x.x') }}
- {{ pin_compatible('hdf5', max_pin='x.x.x') }}

These can be completely deleted in run deps once the upstreams have run_exports.

Comment (jakirkham) [reply recipes/mpb/meta.yaml]: Further discussion in [this thread]( https://github.com/conda-forge/mpb-feedstock/pull/2#discussion_r184384413 ).

**Pull request merged**